source/material_model/visco_plastic.cc

```diff
@@ -78,11 +78,44 @@ namespace aspect
 
       const std::vector<double> volume_fractions = MaterialUtilities::compute_volume_fractions(in.composition[0], get_volumetric_composition_mask());
 
+      /* The following handles phases in a similar way as in the 'evaluate' function.
+      * Results then enter the calculation of plastic yielding.
+      */
+
+      const double gravity_norm = this->get_gravity_model().gravity_vector(in.position[0]).norm();
+
+      double reference_density;
+      if (this->get_adiabatic_conditions().is_initialized())
+        {
+          reference_density = this->get_adiabatic_conditions().density(in.position[0]);
+        }
+      else
+        {
+          EquationOfStateOutputs<dim> eos_outputs_all_phases (this->n_compositional_fields()+1+phase_function.n_phase_transitions());
+          equation_of_state.evaluate(in, 0, eos_outputs_all_phases);
+          reference_density = eos_outputs_all_phases.densities[0];
+        }
+
+      MaterialUtilities::PhaseFunctionInputs<dim> phase_inputs(in.temperature[0],
+                                                               in.pressure[0],
+                                                               this->get_geometry_model().depth(in.position[0]),
+                                                               gravity_norm*reference_density,
+                                                               numbers::invalid_unsigned_int);
+
+      std::vector<double> phase_function_values(phase_function.n_phase_transitions(), 0.0);
+
+      for (unsigned int j=0; j < phase_function.n_phase_transitions(); j++)
+        {
+          phase_inputs.phase_index = j;
+          phase_function_values[j] = phase_function.compute_value(phase_inputs);
+        }
```

Comment:
Yes exactly!
Can you guard this to only execute if there are phase transitions? E.g. like this:
```
std::vector<double> phase_function_values(phase_function.n_phase_transitions(), 0.0);
if (phase_function.n_phase_transitions() > 0)
  {
   ... here goes all the new code that sets values for phase_function_values
  }
here it continues as before
```

Reply:
This is also addressed